### PR TITLE
Clean up some [InlineArray] use

### PR DESCRIPTION
**Pull request by stephentoub**

### CODE REVIEW

File: src/coreclr/nativeaot/System.Private.CoreLib/src/System/Reflection/ConstructorInvoker.cs [hunk 1]
```diff
@@ -67,8 +67,8 @@ public object Invoke(object? arg1, object? arg2)
             }
 
             StackAllocatedArguments argStorage = default;
-            argStorage._args.Set(0, arg1);
```
Comment (jkotas) [inline src/coreclr/nativeaot/System.Private.CoreLib/src/System/Reflection/ConstructorInvoker.cs:70]: Does the JIT generate exact same or better code for this as before this change?

**Comment by jkotas** [comment]
@MihuBot

File: src/coreclr/nativeaot/System.Private.CoreLib/src/System/Reflection/ConstructorInvoker.cs (hunk 1 above)
Comment (stephentoub) [reply src/coreclr/nativeaot/System.Private.CoreLib/src/System/Reflection/ConstructorInvoker.cs:70]: In my initial testing, yes, it was the same. It's certainly possible with more complicated uses it's not. Let's see what the bot comes back with.

File: src/coreclr/nativeaot/System.Private.CoreLib/src/System/Reflection/DynamicInvokeInfo.cs [hunk 2]
```diff
@@ -891,12 +891,6 @@ public Span<T> AsSpan(int length)
                 Debug.Assert((uint)length <= MaxStackAllocArgCount);
                 return new Span<T>(ref _arg0, length);
             }
```
Comment (jkotas) [inline src/coreclr/nativeaot/System.Private.CoreLib/src/System/Reflection/DynamicInvokeInfo.cs:893]: Can this `AsSpan` method be deleted as well?

File: src/coreclr/nativeaot/System.Private.CoreLib/src/System/Reflection/ConstructorInvoker.cs (hunk 1 above)
Comment (jkotas) [reply src/coreclr/nativeaot/System.Private.CoreLib/src/System/Reflection/ConstructorInvoker.cs:70]: The bot agrees.

**Review by jkotas — APPROVED** [review]

File: src/coreclr/nativeaot/System.Private.CoreLib/src/System/Reflection/DynamicInvokeInfo.cs (hunk 2 above)
Comment (stephentoub) [reply src/coreclr/nativeaot/System.Private.CoreLib/src/System/Reflection/DynamicInvokeInfo.cs:893]: Seems like it:
[SharpLab](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBDAzgWwB8ABAJgEYBYAKGIAYACY8gOgCUBXAOwwEt8YLAJI8YUCAAcAymIBuvMDFwBuGvSatOPfoIDCEfBN4AbMTKjzFKtY2YsAIr2wBzLhFx8wuFvoAmMAEEubGMAT1xea2o1AGYmUgZdGgBvGgZ0pjjiFAYAWXIACgB5YAArGDAMBmwoZzo0BhLyyura8gamiqqa51IOsq7W5xiASjSM1OoM6YYAUV5nAAsMTsrcBggBtYYAXgZ/ADNsDmMMVSmZ9M3mjFwAbToAXV2hunPLq63bu/JnvZ7yO8PtcuvdSH8hqQgZcQWs7jEIT0YtCZgBVXAwAqw24sAK4KQSbBcAooEYjFEAXxo43SxCyOVypGKX1e/RuQ3ajRZPT6XPZSLGF3Skw+cwWy1Wtw2X3We0Ox1OKOm2PuTxePTeNJmKp+iLaSoyOvB6tqUK1ypl8L1wwN6XRmIKBQJRIAPNiAHwjFUjFhSYwKTH1Bik8laqlChhau65GAYRYQXxCQzGAoxuMJpMSYxFCR8CBcbwAOQgIn9XF4XGcI0eWokUF4smwGBgGiQTBy9qdhK4Lsl7oYuG7IwYyXD4Zod1LFcCUCg2FCBQAHNXYgOMFAOC15ksVjKUlq6dL2QB9DUow/Ont9hh4y8FCtVSDcDDDnb9mP4aChXI1XCLEI+LATYwHesAHAwp61EGT48KG1AUkAA==)
I'll do so and we can kick off the bot again.

**Pull request merged**